### PR TITLE
Add screenshot-desktop types

### PR DESCRIPTION
**Pull request by usama8800**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [x] `tslint.json` [should contain](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#linter-tslintjson) `{ "extends": "dtslint/dt.json" }`, and no additional rules.
- [x] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

File: types/screenshot-desktop/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,53 @@
+// Type definitions for screenshot-desktop 1.12
+// Project: https://github.com/baz/foo (Does not have to be to GitHub, but prefer linking to a source code repository rather than to a project website.)
+// Definitions by: Usama Ahsan <https://github.com/usama8800>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+/// <reference types="../node" />
```
Comment (peterblazejewicz) [inline types/screenshot-desktop/index.d.ts:6]:
```suggestion
/// <reference types="node" />
```

File: types/screenshot-desktop/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+ *~ This file should be imported using the CommonJS-style:
+ *~   import x = require('someLibrary');
+ *~
+ *~ Refer to the documentation to understand common
+ *~ workarounds for this limitation of ES6 modules.
+ */
+
+/*~ This declaration specifies that the function
+ *~ is the exported object from the file
+ */
```
Comment (peterblazejewicz) [inline types/screenshot-desktop/index.d.ts:18]:
```suggestion
```
those were comments from template

File: types/screenshot-desktop/index.d.ts [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+export = ScreenshotDesktop;
+
+/*~ This example shows how to have multiple overloads for your function */
```
Comment (peterblazejewicz) [inline types/screenshot-desktop/index.d.ts:21]:
```suggestion
```
same, template content

File: types/screenshot-desktop/index.d.ts [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+declare function ScreenshotDesktop(options?: { format?: ScreenshotDesktop.ImageFormat, screen?: ScreenshotDesktop.DisplayID }): Promise<Buffer>;
+declare function ScreenshotDesktop(options?: { filename: string, format?: ScreenshotDesktop.ImageFormat, screen?: ScreenshotDesktop.DisplayID }): Promise<string>;
+
+/*~ If you want to expose types from your module as well, you can
```
Comment (peterblazejewicz) [inline types/screenshot-desktop/index.d.ts:25]:
```suggestion
```
Same

File: types/screenshot-desktop/index.d.ts [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+        'emf' |
+        'exif' |
+        'jpg' |
+        'jpeg' |
+        'gif' |
+        'png' |
+        'tiff' |
+        'wmf';
+
+    /*~ If the module also has properties, declare them here. For example,
```
Comment (peterblazejewicz) [inline types/screenshot-desktop/index.d.ts:44]:
```suggestion
```

File: types/screenshot-desktop/index.d.ts [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+     *~ this declaration says that this code is legal:
+     *~   import f = require('myFuncLibrary');
+     *~   console.log(f.defaultName);
+     */
```
Comment (peterblazejewicz) [inline types/screenshot-desktop/index.d.ts:48]:
```suggestion
```

File: types/screenshot-desktop/index.d.ts [hunk 7]
```diff
@@ -0,0 +1,53 @@
+// Type definitions for screenshot-desktop 1.12
+// Project: https://github.com/baz/foo (Does not have to be to GitHub, but prefer linking to a source code repository rather than to a project website.)
```
Comment (peterblazejewicz) [inline types/screenshot-desktop/index.d.ts:2]: This is wrong, should point to package project page or source repository:
https://github.com/bencevans/screenshot-desktop

File: types/screenshot-desktop/index.d.ts [hunk 8]
```diff
@@ -0,0 +1,53 @@
...
+ *~ workarounds for this limitation of ES6 modules.
+ */
+
+/*~ This declaration specifies that the function
+ *~ is the exported object from the file
+ */
+export = ScreenshotDesktop;
+
+/*~ This example shows how to have multiple overloads for your function */
+declare function ScreenshotDesktop(options?: { format?: ScreenshotDesktop.ImageFormat, screen?: ScreenshotDesktop.DisplayID }): Promise<Buffer>;
```
Comment (peterblazejewicz) [inline types/screenshot-desktop/index.d.ts:22]:
```ts
declare function ScreenshotDesktop(options?: { format?: ScreenshotDesktop.ImageFormat, screen?: ScreenshotDesktop.DisplayID }): Promise<Buffer>;
```
I'd suggest to use JS conventions, where CamelCase is for class or constructor function, say `screenshot` (like in the Readme) or `screenshotDesktop`

File: types/screenshot-desktop/index.d.ts [hunk 9]
```diff
@@ -0,0 +1,31 @@
+// Type definitions for screenshot-desktop 1.12
+// Project: https://github.com/bencevans/screenshot-desktop (Does not have to be to GitHub, but prefer linking to a source code repository rather than to a project website.)
```
Comment (peterblazejewicz) [inline types/screenshot-desktop/index.d.ts:2]:
```suggestion
// Project: https://github.com/bencevans/screenshot-desktop
```

File: types/screenshot-desktop/index.d.ts [hunk 10]
```diff
@@ -0,0 +1,31 @@
...
+        'exif' |
+        'jpg' |
+        'jpeg' |
+        'gif' |
+        'png' |
+        'tiff' |
+        'wmf';
+
+    const defaultName: string;
+    let defaultLength: number;
```
Comment (peterblazejewicz) [inline types/screenshot-desktop/index.d.ts:28]: where those two come from in the native package?

Comment (usama8800) [reply types/screenshot-desktop/index.d.ts:28]: Forgot to remove these.They were part of dts-gen

**Review by DanielRosenwasser — APPROVED** [review]

**Review by DanielRosenwasser — CHANGES_REQUESTED** [review]
actually would rather get @peterblazejewicz's feedback first

**Comment by usama8800** [comment]
I don't get what the requested change is

**Review by peterblazejewicz — APPROVED** [review]
LGMT!

**Review by DanielRosenwasser — APPROVED** [review]

**Comment by usama8800** [comment]
Ready to merge

**Pull request merged**